Rosetta test sometimes fails due to data race

### Description

The rosetta test case `TestBlockServiceSuite/TestBlockWithAccountAlias` fails occasionally:

```
2022-08-18T16:40:14.5349422Z [INFO] ==================
2022-08-18T16:40:14.5349560Z [INFO] WARNING: DATA RACE
2022-08-18T16:40:14.5349727Z [INFO] Write at 0x00c00084c030 by goroutine 68:
2022-08-18T16:40:14.5350203Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockServiceSuite).TestBlockWithAccountAlias.func2()
2022-08-18T16:40:14.5350642Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service_test.go:215 +0x21c
2022-08-18T16:40:14.5350891Z [INFO]   github.com/stretchr/testify/mock.(*Mock).MethodCalled()
2022-08-18T16:40:14.5351344Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/~/go/pkg/mod/github.com/stretchr/testify@v1.8.0/mock/mock.go:538 +0x118e
2022-08-18T16:40:14.5351563Z [INFO]   github.com/stretchr/testify/mock.(*Mock).Called()
2022-08-18T16:40:14.5351993Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/~/go/pkg/mod/github.com/stretchr/testify@v1.8.0/mock/mock.go:450 +0x1a4
2022-08-18T16:40:14.5352402Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/test/mocks.(*MockBlockRepository).FindByIdentifier()
2022-08-18T16:40:14.5352820Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/test/mocks/block_repository.go:46 +0x39
2022-08-18T16:40:14.5353267Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*BaseService).RetrieveBlock()
2022-08-18T16:40:14.5353670Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/base_service.go:99 +0x403
2022-08-18T16:40:14.5354055Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockAPIService).Block()
2022-08-18T16:40:14.5354463Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service.go:58 +0xb6
2022-08-18T16:40:14.5354941Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockServiceSuite).TestBlockWithAccountAlias.func5()
2022-08-18T16:40:14.5355357Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service_test.go:231 +0x61
2022-08-18T16:40:14.5355478Z [INFO] 
2022-08-18T16:40:14.5355663Z [INFO] Previous write at 0x00c00084c030 by goroutine 67:
2022-08-18T16:40:14.5356106Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockServiceSuite).TestBlockWithAccountAlias.func2()
2022-08-18T16:40:14.5356543Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service_test.go:215 +0x21c
2022-08-18T16:40:14.5356788Z [INFO]   github.com/stretchr/testify/mock.(*Mock).MethodCalled()
2022-08-18T16:40:14.5357207Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/~/go/pkg/mod/github.com/stretchr/testify@v1.8.0/mock/mock.go:538 +0x118e
2022-08-18T16:40:14.5357417Z [INFO]   github.com/stretchr/testify/mock.(*Mock).Called()
2022-08-18T16:40:14.5357830Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/~/go/pkg/mod/github.com/stretchr/testify@v1.8.0/mock/mock.go:450 +0x1a4
2022-08-18T16:40:14.5358234Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/test/mocks.(*MockBlockRepository).FindByIdentifier()
2022-08-18T16:40:14.5358645Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/test/mocks/block_repository.go:46 +0x39
2022-08-18T16:40:14.5359044Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*BaseService).RetrieveBlock()
2022-08-18T16:40:14.5359447Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/base_service.go:99 +0x403
2022-08-18T16:40:14.5359815Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockAPIService).Block()
2022-08-18T16:40:14.5360217Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service.go:58 +0xb6
2022-08-18T16:40:14.5360685Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockServiceSuite).TestBlockWithAccountAlias.func4()
2022-08-18T16:40:14.5361099Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service_test.go:230 +0x61
2022-08-18T16:40:14.5361220Z [INFO] 
2022-08-18T16:40:14.5361376Z [INFO] Goroutine 68 (running) created at:
2022-08-18T16:40:14.5361815Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockServiceSuite).TestBlockWithAccountAlias()
2022-08-18T16:40:14.5362228Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service_test.go:231 +0xef3
2022-08-18T16:40:14.5362364Z [INFO]   runtime.call16()
2022-08-18T16:40:14.5362763Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/runtime/asm_amd64.s:701 +0x48
2022-08-18T16:40:14.5362932Z [INFO]   reflect.Value.Call()
2022-08-18T16:40:14.5363333Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/reflect/value.go:339 +0xd7
2022-08-18T16:40:14.5363575Z [INFO]   github.com/stretchr/testify/suite.Run.func1()
2022-08-18T16:40:14.5364009Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/~/go/pkg/mod/github.com/stretchr/testify@v1.8.0/suite/suite.go:175 +0x6dc
2022-08-18T16:40:14.5364145Z [INFO]   testing.tRunner()
2022-08-18T16:40:14.5364555Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/testing/testing.go:1439 +0x213
2022-08-18T16:40:14.5364698Z [INFO]   testing.(*T).Run.func1()
2022-08-18T16:40:14.5365098Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/testing/testing.go:1486 +0x47
2022-08-18T16:40:14.5365236Z [INFO] 
2022-08-18T16:40:14.5365389Z [INFO] Goroutine 67 (running) created at:
2022-08-18T16:40:14.5365830Z [INFO]   github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services.(*blockServiceSuite).TestBlockWithAccountAlias()
2022-08-18T16:40:14.5366244Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/hedera-mirror-rosetta/app/services/block_service_test.go:230 +0xb99
2022-08-18T16:40:14.5366380Z [INFO]   runtime.call16()
2022-08-18T16:40:14.5366770Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/runtime/asm_amd64.s:701 +0x48
2022-08-18T16:40:14.5366914Z [INFO]   reflect.Value.Call()
2022-08-18T16:40:14.5367358Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/reflect/value.go:339 +0xd7
2022-08-18T16:40:14.5367568Z [INFO]   github.com/stretchr/testify/suite.Run.func1()
2022-08-18T16:40:14.5368015Z [INFO]       /home/runner/work/hedera-mirror-node/hedera-mirror-node/~/go/pkg/mod/github.com/stretchr/testify@v1.8.0/suite/suite.go:175 +0x6dc
2022-08-18T16:40:14.5368153Z [INFO]   testing.tRunner()
2022-08-18T16:40:14.5368553Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/testing/testing.go:1439 +0x213
2022-08-18T16:40:14.5368704Z [INFO]   testing.(*T).Run.func1()
2022-08-18T16:40:14.5369215Z [INFO]       /home/runner/.m2/repository/com/igormaznitsa/mvn-golang-wrapper/store/go1.18.linux-amd64/src/testing/testing.go:1486 +0x47
2022-08-18T16:40:14.5369342Z [INFO] ==================
2022-08-18T16:40:14.5369541Z [INFO]     block_service_test.go:239: PASS:	GetAccountAlias()
2022-08-18T16:40:14.5369767Z [INFO]     block_service_test.go:239: PASS:	FindByIdentifier()
2022-08-18T16:40:14.5369965Z [INFO]     block_service_test.go:239: PASS:	FindBetween()
2022-08-18T16:40:14.5500050Z [INFO]     testing.go:1312: race detected during execution of test
2022-08-18T16:40:14.5500382Z [INFO] === RUN   TestBlockServiceSuite/TestNewBlockAPIService
2022-08-18T16:40:14.5500509Z [INFO] === CONT  TestBlockServiceSuite
2022-08-18T16:40:14.5500666Z [INFO]     testing.go:1312: race detected during execution of test
2022-08-18T16:40:14.5500962Z [INFO] --- FAIL: TestBlockServiceSuite (0.02s)
2022-08-18T16:40:14.5501204Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlock (0.00s)
2022-08-18T16:40:14.5501506Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockThrowsWhenAccountRepoFail (0.00s)
2022-08-18T16:40:14.5501839Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockThrowsWhenFindBetweenFails (0.00s)
2022-08-18T16:40:14.5502152Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockThrowsWhenFindByIdentifierFails (0.00s)
2022-08-18T16:40:14.5502401Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockTransaction (0.00s)
2022-08-18T16:40:14.5502738Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockTransactionThrowsWhenAccountRepoFail (0.00s)
2022-08-18T16:40:14.5503102Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockTransactionThrowsWhenFindByHashInBlockFails (0.00s)
2022-08-18T16:40:14.5503451Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockTransactionThrowsWhenFindByIdentifierFails (0.00s)
2022-08-18T16:40:14.5503742Z [INFO]     --- PASS: TestBlockServiceSuite/TestBlockTransactionWithAccountAlias (0.00s)
2022-08-18T16:40:14.5504184Z [INFO]     --- FAIL: TestBlockServiceSuite/TestBlockWithAccountAlias (0.00s)
2022-08-18T16:40:14.5504439Z [INFO]     --- PASS: TestBlockServiceSuite/TestNewBlockAPIService (0.00s)
```

The issue is in the test case code itself, due to `testify/mock` `RunFn` is not reentrant, so sometimes when multiple goroutines call the mock at the same time, concurrent writes happen to the fields of the mocked method object. 

### Steps to reproduce

refer to the log

### Additional context

_No response_

### Hedera network

other

### Version

v0.63.0-rc3

### Operating system

_No response_

--- a/hedera-mirror-rosetta/app/services/block_service_test.go
+++ b/hedera-mirror-rosetta/app/services/block_service_test.go
@@ -32,7 +32,6 @@ import (
 	"github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/persistence/domain"
 	"github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/test/mocks"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/suite"
 )
 
@@ -198,45 +197,25 @@ func (suite *blockServiceSuite) TestBlock() {
 
 func (suite *blockServiceSuite) TestBlockWithAccountAlias() {
 	// given:
-	ch := make(chan interface{})
+	exampleTransactions := []*types.Transaction{
+		makeTransaction(nil, "123"),
+		makeTransaction(&entityId, "246"),
+	}
 	expected := expectedBlockResponse(
 		expectedTransaction(accountAlias, nil, "123"),
 		expectedTransaction(accountAlias, &entityId, "246"),
 	)
-	mockGetAccountAlias := suite.mockAccountRepo.On("GetAccountAlias")
-	mockGetAccountAlias.RunFn = func(_ mock.Arguments) {
-		<-ch // Blocks until the channel is closed
-		mockGetAccountAlias.ReturnArguments = mock.Arguments{accountAlias, mocks.NilError}
-	}
-
-	mockFindByIdentifier := suite.mockBlockRepo.On("FindByIdentifier")
-	mockFindByIdentifier.RunFn = func(_ mock.Arguments) {
-		// Avoid race detector reports concurrent writes on the same returned block struct
-		mockFindByIdentifier.ReturnArguments = mock.Arguments{block(), mocks.NilError}
-	}
-
-	mockFindBetween := suite.mockTransactionRepo.On("FindBetween")
-	mockFindBetween.RunFn = func(_ mock.Arguments) {
-		transactions := []*types.Transaction{
-			makeTransaction(nil, "123"),
-			makeTransaction(&entityId, "246"),
-		}
-		// Avoid race detector reports concurrent writes on the same returned transactions struct
-		mockFindBetween.ReturnArguments = mock.Arguments{transactions, mocks.NilError}
-	}
+	suite.mockAccountRepo.On("GetAccountAlias").Return(accountAlias, mocks.NilError)
+	suite.mockBlockRepo.On("FindByIdentifier").Return(block(), mocks.NilError)
+	suite.mockTransactionRepo.On("FindBetween").Return(exampleTransactions, mocks.NilError)
 
 	// when:
-	// Simulate a race condition
-	go suite.blockService.Block(nil, blockRequest())
-	go suite.blockService.Block(nil, blockRequest())
-	close(ch)
-
-	actual, err := suite.blockService.Block(nil, blockRequest())
+	actual, e := suite.blockService.Block(nil, blockRequest())
 
 	// then:
-	assert.Nil(suite.T(), err)
+	assert.Nil(suite.T(), e)
 	assert.Equal(suite.T(), expected, actual)
-	mock.AssertExpectationsForObjects(suite.T(), suite.mockAccountRepo, suite.mockBlockRepo, suite.mockTransactionRepo)
+	suite.mockAccountRepo.AssertNumberOfCalls(suite.T(), "GetAccountAlias", 1)
 }
 
 func (suite *blockServiceSuite) TestBlockThrowsWhenAccountRepoFail() {

